retis-events/src/python.rs

```diff
@@ -54,67 +56,85 @@ use super::*;
 ///   if 15 (p1_p) 2001:db8:dead::1.20 > 2001:db8:dead::2.80 ttl 64 len 20 proto TCP (6) flags [S] seq 0 win 8192
 /// ```
 #[pyclass(name = "Event")]
-pub struct PyEvent(Event);
+pub struct PyEvent(Py<Event>);
 
 // We need this to make it a pyclass.
 //
 // Safety: no mutable references on internal members is ever given.
 unsafe impl Sync for PyEvent {}
 
 impl PyEvent {
-    pub(crate) fn new(event: Event) -> Self {
-        Self(event)
+    pub(crate) fn new(py: Python<'_>, event: Event) -> PyResult<Self> {
+        Ok(Self(Py::new(py, event)?))
     }
 }
 
 #[pymethods]
 impl PyEvent {
     /// Controls how the PyEvent is represented, eg. what is the output of
     /// `print(e)`.
-    fn __repr__<'a>(&'a self, py: Python<'a>) -> String {
-        let raw = self.raw(py);
-        let dict: &Bound<'_, PyAny> = raw.bind(py);
-        dict.repr().unwrap().to_string()
+    fn __repr__<'a>(&'a self, py: Python<'a>) -> PyResult<Bound<'a, PyString>> {
+        self.0.bind(py).repr()
     }
```

Comment:
Before this PR, printing the event was displaying its raw representation (a dict),

```
>>> print(e)
{'common': {'timestamp': 26713797443116}, 'startup': {'retis_version': 'v1.5.0-c3e05b11767c', 'clock_monotonic_offset': {'sec': 1746429857, 'nsec': 838574294}}}
```

This is the output after this PR,

```
>>> print(e)
<builtins.Event object at 0x5628960da920>
```

(Printing individual sections works as expected).

Reply:
The nice thing about getting a repr of the event as a whole was to quickly see what a given event is about, to of course use other means to inspect it as it is not nicely readable. Having said that your points also make sense. I'm OK with the approach either ways, just make sure the documentation is in sync (haven't checked).

Reply:
Well, I guess we can have this discussion in another PR...